File: easy/26/remove_duplicates_from_sorted_array.ts
function removeDuplicates(nums: number[]): number {
  let k: number = 1;

  for (let i: number = 0; i < nums.length - 1; i++) {
    if (nums[i] !== nums[i + 1]) {
      nums[k] = nums[i + 1];
      k++;
    }
  }

  return k;
}

File: easy/26/remove_duplicates_from_sorted_array.go
package main

import "fmt"

func removeDuplicates(nums []int) int {
	if len(nums) == 0 {
		return 0
	}
	k := 1
	for i := 1; i < len(nums); i++ {
		if nums[k-1] != nums[i] {
			nums[k] = nums[i]
			k++
		}
	}

	return k
}

func main() {
	nums := []int{1, 2, 2, 3, 3, 4, 4, 5, 6, 6, 6}
	k := removeDuplicates(nums)
	fmt.Println(k)
	fmt.Println(nums[:k])
}
